src/backend/distributed/transaction/backend_data.c

```diff
@@ -401,27 +401,39 @@ BackendManagementShmemSize(void)
 
 
 /*
- *  InitializeBackendData is called per backend and does the
- *  required initialization.
+ * InitializeBackendData initialises MyBackendData to the shared memory segment
+ * belonging to the current backend.
+ *
+ * The function is called through CitusHasBeenLoaded when we first detect that
+ * the Citus extension is present, and after any subsequent invalidation of
+ * the metadata.
+ *
+ * We only need to initialise MyBackendData once. The only goal here is to make
+ * sure that we don't use the backend data from a previous backend with the same
+ * pgprocno. Resetting the backend data after a distributed transaction happens
+ * on COMMIT/ABORT through transaction callbacks.
  */
 void
 InitializeBackendData(void)
 {
+	if (MyBackendData != NULL)
```

Comment:
It might have led us skipping some distributed deadlocks :/ 

`InitializeMaintenanceDaemonBackend` already has a similar logic to skip starting the daemon when called for the second time.